File: app/src/main/java/com/silentgames/silent_planet/customListeners/CustomGestureListener.java
package com.silentgames.silent_planet.customListeners;

import android.view.GestureDetector;
import android.view.MotionEvent;

import com.silentgames.silent_planet.logic.Constants;

/**
 * Created by gidroshvandel on 21.06.17.
 */
public class CustomGestureListener extends GestureDetector.SimpleOnGestureListener {

    private Callback callback;

    public interface Callback {

        void onScroll(float distanceX, float distanceY);

        void onSingleTapConfirmed(MotionEvent event);

    }

    public CustomGestureListener(Callback callback) {
        this.callback = callback;
    }

    @Override
    public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
        callback.onScroll(distanceX, distanceY);
        return true;
    }

    @Override
    public boolean onSingleTapConfirmed(MotionEvent event) {
        callback.onSingleTapConfirmed(event);
        return true;
    }
}

File: android/src/com/silentgames/silent_planet/customListeners/CustomGestureListener.kt
package com.silentgames.silent_planet.customListeners

import android.view.GestureDetector.SimpleOnGestureListener
import android.view.MotionEvent

/**
 * Created by gidroshvandel on 21.06.17.
 */
class CustomGestureListener(private val callback: Callback) : SimpleOnGestureListener() {

    interface Callback {
        fun onScroll(distanceX: Float, distanceY: Float)
        fun onSingleTapConfirmed(event: MotionEvent?)
    }

    override fun onScroll(
        e1: MotionEvent,
        e2: MotionEvent,
        distanceX: Float,
        distanceY: Float
    ): Boolean {
        callback.onScroll(distanceX, distanceY)
        return true
    }

    override fun onSingleTapConfirmed(event: MotionEvent): Boolean {
        callback.onSingleTapConfirmed(event)
        return true
    }
}
